packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -904,6 +913,24 @@ export class GarbageCollector implements IGarbageCollector {
 			lastSummaryTime: this.getLastSummaryTimestampMs(),
 			viaHandle: requestHeaders?.[RuntimeHeaders.viaHandle],
 		});
+
+		// We may throw when loading an Inactive object, depending on these preconditions
```

Comment:
You can return here if `reason !== "Loaded"` and avoid running any of the code below.